zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/CorrelatedMessageFromProcessMessageSubscriptionHandler.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.handlers;
+
+import static io.camunda.exporter.utils.ExporterUtil.tenantOrDefault;
+import static io.camunda.exporter.utils.ExporterUtil.toOffsetDateTime;
+
+import io.camunda.exporter.store.BatchRequest;
+import io.camunda.webapps.schema.entities.CorrelatedMessageEntity;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import io.camunda.zeebe.protocol.record.intent.ProcessMessageSubscriptionIntent;
+import io.camunda.zeebe.protocol.record.value.ProcessMessageSubscriptionRecordValue;
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+
+public record CorrelatedMessageFromProcessMessageSubscriptionHandler(String indexName)
```

Comment:
🔧 This is the simplest approach and it's good enough, but, do you think it make sense to create an abstract class to centralise the common behaviour ?

Reply:
I was thinking about it, but felt like it's not worth it. In the end, the records where the data comes from are different, so there are only very few things to generalize, like the ID creation and some record fields.

But maybe that's still helpful, so the ID creation is fixed. I'll do it 👍

Reply:
Done with https://github.com/camunda/camunda/pull/37500/commits/d15aaa027300d0be46a302dd5c3d48ee741f41b4